enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -104,198 +156,160 @@ This is often required for industries such as the US public sector, healthcare o
-* Log forwarding is enabled for tech preview when the instance of `clusterloggging` is annotated with: `clusterlogging.openshift.io/logforwardingtechpreview: enabled`
+* Forwarder CRs can refer to secrets for TLS certificates (see security section below)
+* The operator allows multiple instances of `forwarder`.
+  * Instances in the `openshift-logging` namespace are "global" and can forward logs from any container.
+  * Instances in other namespaces are namespace-restricted and can only forward logs from their own namespace.
+* Log forwarding is enabled when a `forwarder` object is created, and disabled when no forwarder objects exist.
+* When log forwarding is disabled, default forwarding to the internal elasticsearch instance is enabled.
+* When log forwarding is enabled, default forwarding is disabled. To enable both, create a global forwarder in the openshift-logging namespace with a single pipeline: `pipeline: { output: openshift-logging-default }`
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
+* An output MAY support basic server-authenticated TLS by using a URL scheme that requires TLS, e.g. elasticsearch+https:// syslog+tls:// etc.
+* An output MAY support client-authenticated TLS by declaring a `secret` with keys `tls.crt`, `tls.key`, `ca.crt`
+* An output may support username/password or shared-key authentication either in the `url` or in the `secret`.
+* Outputs that intentionally do not use TLS must have an explicit `insecure: true` to avoid security breaches due to mis-configuration.
 
 #### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
-
-For example, If you have specified a pipeline for `logs.app` sources, but did not specify one for `logs.infra`, then `logs.infra` will be dropped
-
+* Pipelines can select multiple container log streams (fan in) by:
+  * Source type (`Infrastructure`, `Application`, `Audit`) 
+  * k8s label selector map or expressions.
+  * namespace (for global forwarders)
+* Selected log streams are routed to all of the specified outputs (fan out)
 
 ### Risks and Mitigations
 
 ## Design Details
 
 The `cluster-logging-operator` will use the forwarding API to generate a collector configuration and mount any secrets into the collector pod.
 
-### API Specification
-The log forwarding API:
+Although there can be many `forwarder` objects, there is still only one collector. The operator effectively joins the configurations from all the forwarders to generate a single collector configuration, and restricts log streams to namespace-restricted forwarders.
```

Comment:
given the presumed difficulties of merging/de-duping configurations, is it reasonable for GA to only support global forwarding, and introduce namespaced forwarding as a follow on?

Reply:
You read our minds.